File: frontend/src/types/campaign.ts
// Types for the Campaign Workflow feature

// Possible status values for a campaign task
export type TaskStatus = 'Pending' | 'In Progress' | 'Needs Review' | 'Approved' | 'Posted' | 'Error';

// Structure of a single campaign task
export interface CampaignTask {
  id: string;
  taskType: string; // e.g., 'repurpose', 'create_image_prompt'
  targetFormat?: string; // e.g., 'LinkedIn Post', 'Tweet Thread'
  targetAsset?: string;  // e.g., 'Blog Header', 'LinkedIn Post Image'
  status: TaskStatus;
  result?: string | null; // The generated asset or null
  imageUrl?: string;    // For image assets, if available
  error?: string | null;
  createdAt: string;
  updatedAt: string;
}

// Structure of the campaign plan (list of tasks)
export type CampaignPlan = CampaignTask[];

// Structure of the campaign response from the backend
export interface CampaignResponse {
  id: string;
  blogId: string;
  createdAt: string;
  tasks: CampaignTask[];
} 

File: src/api/models/campaign.py
"""Campaign-related API models."""

from pydantic import BaseModel
from typing import List, Optional


class CampaignCreateRequest(BaseModel):
    blog_id: str


class CampaignTaskExecuteRequest(BaseModel):
    task_id: str


class CampaignTaskUpdateRequest(BaseModel):
    content: Optional[str] = None
    status: str


class CampaignTaskResponse(BaseModel):
    id: str
    task_type: str
    target_format: Optional[str]
    target_asset: Optional[str]
    status: str
    result: Optional[str]
    image_url: Optional[str]
    error: Optional[str]
    created_at: str
    updated_at: str


class CampaignResponse(BaseModel):
    id: str
    blog_id: str
    created_at: str
    tasks: List[CampaignTaskResponse]